docs/sdk/src/reference_docs/chain_spec_genesis.rs

```diff
@@ -1,4 +1,110 @@
 //! Chain spec and genesis build.
 //!
-//! What is chain-spec.
-//! What is genesis state and how to build it.
+//! # What is chain-spec.
+//!
+//! Chain specicfication file defines the set of properties that are required to run the node as
+//! part of the chain. The chain specification consists of two main parts:
+//! - initial state of the runtime,
+//! - network / logical properties of the chain, the most important property being the list of
+//!   bootnodes.
+//!
+//! This document describes how initial state is handled in pallets and runtime, and how to interact
+//! with runtime in order to build genesis sate.
+//!
+//! For more informations on chain specification and its properties refer to
+//! [`sc_chain_spec#from-initial-state-to-raw-genesis`].
+//!
+//! The initial genesis state can be provided in the following formats:
+//!   - full,
+//!   - patch,
+//!   - raw,
+//!
+//! Each of the format is explained in [_chain-spec-format_][`sc_chain_spec#chain-spec-formats`].
+//!
+//!
+//! ## `GenesisConfig` for `pallet`
+//!
+//! Every frame pallet may have its initial state which is defined by the `GenesisConfig` internal
+//! struct. It is a regular rust struct, annotated with [`pallet::genesis_config`] attribute.
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_bar_GenesisConfig)]
+
+//! The struct shall be defined within the pallet `mod`, as in the following code:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_bar)]
+
+//! The initial state conveyed in `GenesisConfig` struct is transformed into state storage items by
+//! means of the [`BuildGenesisConfig`] trait wich shall be implemented for pallet's `GenesisConfig`
+//! struct. The [`pallet::genesis_build`] attribute shall be attached to the `impl` block:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_bar_build)]
+
+//! GenesisConfig may contain also more complicated types including nested structs or enums, as in
+//! the example for `pallet_foo`:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_foo_GenesisConfig)]
+
+//! Please note that fields of `GenesisConfig` may not be directly mapped to storage items:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_foo_build)]
+
+//! ## `GenesisConfig` for `runtimes`
+
+//! The runtime genesis config struct consists of configs for every pallet. For [_demonstration
+//! runtime_][`chain_spec_guide_runtime`] used in this guide, it consists of `SystemConfig`,
+//! `BarConfig` and `FooConfig`. This structure was automatically generated by macro and it can be
```

Comment:
I can not find `SystemConfig`, `BarConfig` and `FooConfig` anywhere, maybe leftovers from an older version of this doc?

Reply:
Macro generated names, just click the link provided for sneak peaking...